cmd/storagenode-updater/main.go

```diff
@@ -98,9 +113,13 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 	if err != nil {
 		log.Fatalf("error loading identity: %s", err)
 	}
+	nodeID = ident.ID
+	if nodeID.IsZero() {
+		log.Fatal("empty node ID")
```

Comment:
The command should't return an error imo. I forgot to remove the return type.

Reply:
:thinking: I've seen recently in other binaries that for command functions the signature of them is to return an error.

I'm not currently saying which one is better, only mentioning what I've seen and unless that you are sure about it, I would ask which form is the more homogeneous.

Reply:
I need to return error because its registered with `RunE:  cmdRun,` and our integration with cobra forces us to use `RunE`.

Reply:
@mniewrzal thanks for asserting it, I was thinking that we are using that form, but I wasn't 100% sure which form we prefer.

Reply:
But going back to initial comment. I think returning error here will be less convenient because then error will be wrapped in command help content. It will display usage + error + all flags (as far I remember).

Reply:
I did a run returning an error and calling `fatal` and these are the outputs:

__Returning error__

```
2019-10-29T18:26:15.818+0100    INFO    process/exec_conf.go:240        Configuration loaded from: /hostmachine/workspace/projects/storj/releases/dev/v0.22.2/config/uplink/config.yaml
2019-10-29T18:26:15.818+0100    INFO    process/exec_conf.go:250        Invalid configuration file key: scope                                                         
2019-10-29T18:26:15.818+0100    INFO    process/exec_conf.go:254        Invalid configuration file value for key: log.level                                           
2019-10-29T18:26:15.818+0100    FATAL   process/exec_conf.go:299        Unrecoverable error     {"error": "haha"}                                                     
storj.io/storj/pkg/process.cleanup.func1
        /hostmachine/workspace/projects/storj/repos/storj/pkg/process/exec_conf.go:299                                                                                
github.com/spf13/cobra.(*Command).execute
        /home/ivan/persistent/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826                                                                                 
github.com/spf13/cobra.(*Command).ExecuteC
        /home/ivan/persistent/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914                                                                                 
github.com/spf13/cobra.(*Command).Execute
        /home/ivan/persistent/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864                                                                                 
storj.io/storj/pkg/process.Exec
        /hostmachine/workspace/projects/storj/repos/storj/pkg/process/exec_conf.go:74                                                                                 
main.main
        /hostmachine/workspace/projects/storj/repos/storj/cmd/storagenode-updater/main.go:381                                                                         
runtime.main
        /apps/go/src/runtime/proc.go:203
exit status 1
```

__Calling `log.Fatal`__

```
2019-10-29T18:27:31.779+0100    INFO    process/exec_conf.go:240        Configuration loaded from: /hostmachine/workspace/projects/storj/releases/dev/v0.22.2/config/uplink/config.yaml
2019-10-29T18:27:31.779+0100    INFO    process/exec_conf.go:250        Invalid configuration file key: scope                                                         
2019-10-29T18:27:31.779+0100    INFO    process/exec_conf.go:254        Invalid configuration file value for key: log.output                                          
2019-10-29T18:27:31.779+0100    INFO    storagenode-updater/main.go:100 haha
exit status 1
```

The error help message isn't shown but returning an error returns a fatal message log a part of showing the ugly stack trace.